src/app/util/af-types.h

```diff
@@ -124,15 +125,7 @@ struct EmberAfCluster
     bool IsClient() const { return (mask & CLUSTER_MASK_CLIENT) != 0; }
 };
 
-/**
- * @brief Struct that represents a logical device type consisting
- * of a DeviceID and its version.
- */
-typedef struct
-{
-    chip::DeviceTypeId deviceId;
-    uint8_t deviceVersion;
-} EmberAfDeviceType;
+using EmberAfDeviceType = chip::app::DataModel::DeviceTypeEntry;
```

Comment:
This is breaking public API, no?  Because the structs have different field names....

Why is that OK?

Reply:
I did not think this was that much of a public API: we seemed to mainly use it to expose ember bits and not in other code (i.e. codegen from ember goes into something that ends up into matter objects).

If we care about names, I can keep the old names. However `deviceVersion` seems to be for what the spec calls `revision history` so it is not a great name ... it seems more inherited han intentional.